sentinel/pkg/checker/dal/app.go

```diff
@@ -110,6 +163,98 @@ func checkDal(endpoint string, key string, alarmCount map[string]int) error {
 	return nil
 }
 
+func checkDalWs(ctx context.Context) {
+	log.Debug().Msg("checking msg delays")
+	msgs := extractWsAlarms(ctx)
+	if len(msgs) > 0 {
+		alert.SlackAlert(strings.Join(msgs, "\n"))
+	}
+
+	log.Debug().Msg("checking msg push")
+	msgsNotRecieved := []string{}
+	updateTimes.Range(func(key, value interface{}) bool {
+		if recievedTime, ok := value.(time.Time); ok {
+			diff := time.Since(recievedTime)
+			if diff > 1*time.Second {
+				symbol := key.(string)
+				msg := fmt.Sprintf("(%s) ws not pushed for %v(sec)", symbol, diff.Seconds())
+				msgsNotRecieved = append(msgsNotRecieved, msg)
+			}
+		}
+		return true
+	})
+	if len(msgsNotRecieved) > 0 {
+		alert.SlackAlert(strings.Join(msgsNotRecieved, "\n"))
+	}
+}
+
+func extractWsAlarms(ctx context.Context) []string {
+	var msgs = []string{}
+	select {
+	case <-ctx.Done():
+		return nil
+	case entry := <-wsMsgChan:
+		msgs = append(msgs, entry)
+	loop:
+		for {
+			select {
+			case entry := <-wsMsgChan:
+				msgs = append(msgs, entry)
+			default:
+				break loop
+			}
+		}
+	}
+	return msgs
+}
+
 func checkValueEmptiness(data *OutgoingSubmissionData) bool {
 	return data.Symbol == "" || data.Value == "" || data.AggregateTime == "" || data.Proof == "" || data.FeedHash == "" || data.Decimals == ""
 }
+
+func fetchConfigs() ([]Config, error) {
+	chain := os.Getenv("CHAIN")
+	if chain == "" {
+		log.Info().Str("Player", "Reporter").Msg("CHAIN env not set, defaulting to baobab")
+		chain = "baobab"
+	}
+	endpoint := fmt.Sprintf("https://config.orakl.network/%s_configs.json", chain)
+	configs, err := request.Request[[]Config](request.WithEndpoint(endpoint))
+	if err != nil {
+		return nil, err
+	}
+	return configs, nil
+}
+
+func handleWsMessage(ctx context.Context, data map[string]interface{}) error {
+	wsData := WsResponse{}
+	jsonMarshalData, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(jsonMarshalData, &wsData)
+	if err != nil {
+		return err
+	}
+	defer updateTimes.Store(wsData.Symbol, time.Now())
+	wsChan <- wsData
+	return nil
+}
+
+func filterWsReponses() {
+	for entry := range wsChan {
+		strTimestamp := entry.AggregateTime
+
+		unixTimestamp, err := strconv.ParseInt(strTimestamp, 10, 64)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to parse timestamp")
+			continue
+		}
+
+		timestamp := time.Unix(unixTimestamp, 0)
+		diff := time.Since(timestamp)
+		if diff > 5*time.Second {
```

Comment:
Ideally we want to be subsecond, right? Is this threshold okay?

Reply:
which threshold do you suggest? I was worried if it might be spamming if too low